v0/destinations/recurly/util.js

```diff
@@ -0,0 +1,92 @@
+const btoa = require("btoa");
+const axios = require("axios");
+const logger = require("../../../logger");
+const { ACCEPT_HEADERS, DEFAULT_BASE_ENDPOINT } = require("./config");
+const {
+  ErrorMessage,
+  isDefinedAndNotNullAndNotEmpty,
+  getValueFromMessage
+} = require("../../util");
+
+const createItem = async (data, config, relativePath) => {
+  let response = null;
+  const itemUrl = `${DEFAULT_BASE_ENDPOINT}${relativePath}`;
+  try {
+    response = await axios.post(itemUrl, data, {
+      headers: {
+        "Content-Type": "application/json",
+        Accept: ACCEPT_HEADERS,
+        Authorization: `Basic ${btoa(`${config.apiKey}:`)}`
+      }
+    });
+  } catch (err) {
+    logger.debug(ErrorMessage.ObjectNotFound);
+  }
+  if (response && response.status === 201 && response.data.id) {
```

Comment:
```suggestion
  if (response && response.data && response.data.id) {
```

checking with hard-coded status code is riskier IMO.